File: src/main/java/hiiragi283/material/api/machine/MachineContents.java
package hiiragi283.material.api.machine;

import hiiragi283.material.api.capability.fluid.HiiragiFluidTank;
import hiiragi283.material.api.capability.item.HiiragiItemHandler;

public record MachineContents(
        HiiragiItemHandler inputItemHandler,
        HiiragiFluidTank inputTank0,
        HiiragiFluidTank inputTank1,
        HiiragiFluidTank inputTank2,
        IMachineProperty property
) {
}

File: src/main/kotlin/hiiragi283/material/api/machine/MachineContents.kt
package hiiragi283.material.api.machine

import hiiragi283.material.api.capability.fluid.HiiragiFluidTank
import hiiragi283.material.api.capability.item.HiiragiItemHandler

data class MachineContents(
    val inputItemHandler: HiiragiItemHandler,
    val inputTank0: HiiragiFluidTank,
    val inputTank1: HiiragiFluidTank,
    val inputTank2: HiiragiFluidTank,
    val property: IMachineProperty
)